test/features/georouting/georouting.test.js

```diff
@@ -0,0 +1,356 @@
+    expect(text[0].textContent).to.be.equal( mockGeoroutingJson.data.find(d => d.prefix === 'ch_de').text)
+    expect(text[1].textContent).to.be.equal( mockGeoroutingJson.data.find(d => d.prefix === 'ch_fr').text)
+    expect(text[2].textContent).to.be.equal( mockGeoroutingJson.data.find(d => d.prefix === '').text)
+    const links = modal.querySelectorAll('a');
+    expect(links).to.not.be.null;
+    expect(links.length).to.be.equal(3);
+    expect(links[0].text).to.be.equal( mockGeoroutingJson.data.find(d => d.prefix === 'ch_de').button)
+    expect(links[1].text).to.be.equal( mockGeoroutingJson.data.find(d => d.prefix === 'ch_fr').button)
+    expect(links[2].text).to.be.equal( mockGeoroutingJson.data.find(d => d.prefix === '').button)
+    // cleanup
+    stubFallbackMetadata('on');
+    stubHeadRequestToReturnVal('/ch_it', true);
+  });
+
+  it('Will show fallback links if fallbackrouting is on and page exist request fails', async () => {
+    // prepare
+    stubHeadRequestToReturnVal('/ch_it', false);
+    await init(mockConfig, createTag, getMetadata);
+    const modal = document.querySelector('.dialog-modal');
+    // assert
+    expect(modal).to.not.be.null;
+    const text = modal.querySelectorAll('.locale-text');
+    expect(text).to.not.be.null;
+    expect(text.length).to.be.equal(4);
+    const links = modal.querySelectorAll('a');
+    expect(links).to.not.be.null;
+    expect(links.length).to.be.equal(4);
+    // cleanup
+    stubHeadRequestToReturnVal('/ch_it', true);
+  });
+
+  it('Will not display modal if no single link was found and if fallback routing is off', async () => {
+    // prepare
+    stubFallbackMetadata('off');
+    stubHeadRequestToReturnVal('/ch_de', false);
+    stubHeadRequestToReturnVal('/ch_it', false);
+    stubHeadRequestToReturnVal('/ch_fr', false);
+    await init(mockConfig, createTag, getMetadata);
+    const modal = document.querySelector('.dialog-modal');
+    // assert
+    expect(modal).to.be.null;
+    // cleanup
+    stubFallbackMetadata('on');
+    stubHeadRequestToReturnVal('/ch_de', true);
+    stubHeadRequestToReturnVal('/ch_it', true);
+    stubHeadRequestToReturnVal('/ch_fr', true);
+  });
+
+  it('Sets international and georouting_presented cookies on link click in modal', async () => {
+    // prepare
+    await init(mockConfig, createTag, getMetadata);
+    const modal = document.querySelector('.dialog-modal');
+    let cookie = getCookie('international');
+    let storage = sessionStorage.getItem('international');
+    // assert
+    expect(modal).to.not.be.null;
+    expect(cookie).to.be.undefined;
+    expect(storage).to.be.null;
+    const links = modal.querySelectorAll('a');
+    expect(links).to.not.be.null;
+    expect(links[0].text).to.be.equal( mockGeoroutingJson.data.find(d => d.prefix === 'ch_de').button);
+    links[3].click();
+    expect(getCookie('international')).to.be.equal('us');
+    expect(sessionStorage.getItem('international')).to.be.equal('us');
+  });
+});
```

Comment:
All these expects seem excessive, no? Seems a little weird the tests are double the size of the feature.

Reply:
Maybe some are a bit excessive, but for the majority I've put the expect there for a reason. There's a lot of different test cases I'm handling, which is why the test file has grown so large. The way I see it, I'd rather have too many, than too little tests, especially for something this "critical".